lib/cldr/validate_upstream_assumptions.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Cldr
+  class ValidateUpstreamAssumptions
+    class << self
+      def validate
+        errors = []
+        errors += validate_aliases_only_in_root_locale
+        errors += validate_aliases_only_in_expected_paths
+        errors.each do |error|
+          puts(error.message)
+        end
+        exit(errors.empty? ? 0 : 1)
+      end
+
+      private
+
+      def validate_aliases_only_in_root_locale
+        errors = []
+        Cldr::Export.minimum_draft_status = Cldr::DraftStatus::CONTRIBUTED
+        Cldr::Export::Data::RAW_DATA.locales.each do |locale|
+          aliases = Cldr::Export::Data::RAW_DATA[locale].xpath("//alias")
+          unless aliases.empty? || locale == :root
+            errors << StandardError.new("`#{locale}` has aliases but is not `root`")
+          end
+        end
+        errors
+      end
+
+      EXPECTED_ALIAS_LOCATIONS = [
+        "/ldml/dates/calendars/calendar/dayPeriods/dayPeriodContext/dayPeriodWidth/alias",
+        "/ldml/dates/calendars/calendar/days/dayContext/dayWidth/alias",
+        "/ldml/dates/calendars/calendar/months/monthContext/monthWidth/alias",
+        "/ldml/dates/calendars/calendar/quarters/quarterContext/quarterWidth/alias",
+      ].freeze
+
+      def validate_aliases_only_in_expected_paths
+        errors = []
+        Cldr::Export.minimum_draft_status = Cldr::DraftStatus::CONTRIBUTED
+        alias_paths = Cldr::Export::Data::RAW_DATA[:root].xpath("//alias").map do |child|
+          child.path.gsub(/\[\d+\]/, "")
+        end.sort.uniq
+
+        unknown_alias_paths = alias_paths - EXPECTED_ALIAS_LOCATIONS
+
+        unknown_alias_paths.each do |path|
+          errors << StandardError.new("Found an alias at unknown location `#{path}`; Make sure ruby-cldr supports it, then update EXPECTED_ALIAS_LOCATIONS")
+        end
+        errors
```

Comment:
It may be worth formatting the list of unknown locations to improve readability:

```suggestion
        unknown_paths_string = +""
        unknown_alias_paths.each do |path|
          unknown_paths_string << "\n\t#{path}"
        end
        errors << StandardError.new("Found an alias at the following unknown locations: \n (Make sure ruby-cldr supports them, then update EXPECTED_ALIAS_LOCATIONS) #{unknown_paths_string}") unless unknown_paths_string.empty?
```

```
Found an alias at the following unknown locations: 
 (Make sure ruby-cldr supports them, then update EXPECTED_ALIAS_LOCATIONS) 
        /ldml/dates/calendars/calendar/cyclicNameSets/alias
        /ldml/dates/calendars/calendar/cyclicNameSets/cyclicNameSet/alias
        /ldml/dates/calendars/calendar/cyclicNameSets/cyclicNameSet/cyclicNameContext/cyclicNameWidth/alias
        /ldml/dates/calendars/calendar/dateFormats/alias
        /ldml/dates/calendars/calendar/dateTimeFormats/alias
        /ldml/dates/calendars/calendar/dayPeriods/alias
        /ldml/dates/calendars/calendar/days/alias
        /ldml/dates/calendars/calendar/eras/alias
        /ldml/dates/calendars/calendar/eras/eraNames/alias
        /ldml/dates/calendars/calendar/eras/eraNarrow/alias
        /ldml/dates/calendars/calendar/monthPatterns/alias
        /ldml/dates/calendars/calendar/monthPatterns/monthPatternContext/monthPatternWidth/alias
        /ldml/dates/calendars/calendar/months/alias
        /ldml/dates/calendars/calendar/quarters/alias
        /ldml/dates/calendars/calendar/timeFormats/alias
        /ldml/dates/fields/field/alias
        /ldml/listPatterns/listPattern/alias
        /ldml/numbers/currencyFormats/alias
        /ldml/numbers/currencyFormats/currencyFormatLength/currencyFormat/alias
        /ldml/numbers/currencyFormats/currencySpacing/alias
        /ldml/numbers/decimalFormats/alias
        /ldml/numbers/decimalFormats/decimalFormatLength/alias
        /ldml/numbers/miscPatterns/alias
        /ldml/numbers/percentFormats/alias
        /ldml/numbers/scientificFormats/alias
        /ldml/numbers/symbols/alias
        /ldml/units/unitLength/alias
        /ldml/units/unitLength/unit/alias
```